### ISSUE TITLE
react-native-keychain

### ISSUE THREAD

**Issue by nicolasburtey** [GaloyMoney/galoy-mobile#848]
it seems to no longer be used

**Comment by 2624789** [comment]
@nicolasburtey I've been going through the app authentication logic and also looking where is keychain already used.

I saw that [app/utils/storage/secureStorage.ts](https://github.com/GaloyMoney/galoy-mobile/blob/main/app/utils/storage/secureStorage.ts) uses [react-native-secure-key-store](https://github.com/pradeep1991singh/react-native-secure-key-store) but [DeviceAccountModal](https://github.com/GaloyMoney/galoy-mobile/blob/main/app/screens/get-started-screen/device-account-modal.tsx) is using [react-native-keychain](https://www.npmjs.com/package/react-native-keychain).

I would lean to use [react-native-keychain](https://www.npmjs.com/package/react-native-keychain) because [react-native-secure-key-store](https://github.com/pradeep1991singh/react-native-secure-key-store) last update was two years ago. Although, [react-native-keychain](https://www.npmjs.com/package/react-native-keychain) use is limited to the [DeviceAccountModal](https://github.com/GaloyMoney/galoy-mobile/blob/main/app/screens/get-started-screen/device-account-modal.tsx) component, while [react-native-secure-key-store](https://github.com/pradeep1991singh/react-native-secure-key-store) is used more broadly in the app through the [KeyStoreWrapper](https://github.com/GaloyMoney/galoy-mobile/blob/main/app/utils/storage/secureStorage.ts).

I would like to know your opinion about which library would be a better fit to store the token using keychain. Do you think it is a good idea to extend the existing **KeyStoreWrapper** and keep using **react-native-secure-key-store**? Another alternative would be to refactor **KeyStoreWrapper** to use **react-native-keychain** and unify the use of keychain in the app through this class.

**Comment by nicolasburtey** [comment]
another key aspect to keep in mind is to the upgrade path for existing user. we don't want them to having to log back in when we do this upgrade

**Comment by 2624789** [comment]
I agree that for maintainability it is better to have just one library. I'll focus on the feature of this issue that is storing the token on the keychain/keystore without users having to log back. The easiest way I find now for doing this would be to use [react-native-secure-key-store](https://github.com/pradeep1991singh/react-native-secure-key-store).

Removing one of the two libraries and improve maintainability could be a good task for another issue.

**Comment by designsats** [comment]
Resolved — in two parts, both landed.

The original report (an unused dependency) was fixed by #883, merged 2023-01-16, four days after this was filed: `app/utils/keychain.ts` held a `KeychainAction` wrapper that nothing imported, and that PR deleted the wrapper, its jest mock, the `package.json` entry and the pod. `react-native-keychain` came back in #2334 (device account v2) with real callers, and today it is genuinely used by `use-create-device-account.ts` and the self-custodial `use-credential-backup.ts`.

The follow-up raised in the comments — storing the session token in the keychain/keystore without forcing existing users to log back in — was fixed by #4160, merged 2026-08-18. `galoyAuthToken` no longer sits in the plaintext AsyncStorage blob; it lives in a keychain slot via `KeyStoreWrapper`, and a legacy blob token is adopted into the keychain on first load, so no one is logged out by the upgrade.

Covered by regression tests: `adopts a legacy blob token into the keychain and re-saves the blob without it` and `prefers the keychain token over a stale blob token and still scrubs the blob`, in `__tests__/store/persistent-state/index.spec.tsx`.

Still not done, but deliberately scoped out here per the comment above: consolidating `react-native-keychain` and `react-native-secure-key-store` into one library. Worth a separate issue.

**Comment by grimen** [comment]
**Problem**
The app uses two overlapping secure-storage libraries — `react-native-keychain` and `react-native-secure-key-store` — where one would do, which is extra maintenance surface (two native modules, two APIs, two upgrade paths to track).

**Context**
This issue's original report (an unused `react-native-keychain` wrapper) and its direct follow-up (moving the session token into the keychain without forcing re-login) are both already resolved:
- Unused wrapper removed in [#883](https://github.com/blinkbitcoin/blink-mobile/pull/883) (2023-01-16).
- `react-native-keychain` now has real callers: `use-create-device-account.ts` and the self-custodial `use-credential-backup.ts` (added via device-account v2, [#2334](https://github.com/blinkbitcoin/blink-mobile/pull/2334)).
- Session token (`galoyAuthToken`) moved out of plaintext `AsyncStorage` into the keychain via `KeyStoreWrapper` in [#4160](https://github.com/blinkbitcoin/blink-mobile/pull/4160) (merged 2026-08-18), with a legacy-blob-token adoption path so existing users aren't logged out.
- What's left, explicitly scoped out in the thread as its own task: consolidating `react-native-keychain` and `react-native-secure-key-store` into a single library, keeping the same no-forced-logout constraint.

**Acceptance criteria**
- [ ] Decide on one secure-storage library (`react-native-keychain` vs `react-native-secure-key-store`) for the whole app.
- [ ] All secure-storage call sites go through a single wrapper backed by that one library.
- [ ] Existing users are not logged out / do not lose stored credentials as part of the migration (same constraint `KeyStoreWrapper`'s legacy-blob adoption already satisfies for the token).
- [ ] The unused library and its native dependency (pod/Gradle entry) are removed.
- [ ] Covered by regression tests for the migration path, mirroring the existing `persistent-state` adoption tests.
